src/reanimated2/globals.d.ts

```diff
@@ -1,143 +1,96 @@
+  ShadowNodeWrapper,
 } from './commonTypes';
 import type { FrameCallbackRegistryUI } from './frameCallback/FrameCallbackRegistryUI';
-import type { ShadowNodeWrapper } from './hook/commonTypes';
-import { LayoutAnimationStartFunction } from './layoutReanimation';
 import type { NativeReanimated } from './NativeReanimated/NativeReanimated';
+import type {
+  LayoutAnimationFunction,
+  LayoutAnimationType,
+  LayoutAnimationsValues,
+} from './layoutReanimation/animationBuilder';
 
 declare global {
-  const _WORKLET: boolean;
-  const _IS_FABRIC: boolean;
-  const _REANIMATED_VERSION_CPP: string;
-  const _REANIMATED_VERSION_BABEL_PLUGIN: string;
-  const __reanimatedModuleProxy: NativeReanimated;
-  const evalWithSourceMap: (
+  var _WORKLET: boolean;
+  var _IS_FABRIC: boolean;
+  var _REANIMATED_VERSION_CPP: string;
+  var _REANIMATED_VERSION_BABEL_PLUGIN: string;
+  var __reanimatedModuleProxy: NativeReanimated;
+  var evalWithSourceMap: (
     js: string,
     sourceURL: string,
     sourceMap: string
   ) => any;
-  const evalWithSourceUrl: (js: string, sourceURL: string) => any;
-  const _log: (s: string) => void;
-  const _getCurrentTime: () => number;
-  const _notifyAboutProgress: (
+  var evalWithSourceUrl: (js: string, sourceURL: string) => any;
+  var _log: (s: string) => void;
+  var _getCurrentTime: () => number;
+  var _notifyAboutProgress: (
     tag: number,
     value: number,
     isSharedTransition: boolean
   ) => void;
-  const _notifyAboutEnd: (
+  var _notifyAboutEnd: (
     tag: number,
     finished: boolean,
     removeView: boolean
   ) => void;
-  const _setGestureState: (handlerTag: number, newState: number) => void;
-  const _makeShareableClone: (value: any) => any;
-  const _updateDataSynchronously: (
+  var _setGestureState: (handlerTag: number, newState: number) => void;
+  var _makeShareableClone: (value: any) => any;
+  var _updateDataSynchronously: (
     dataHolder: ShareableSyncDataHolderRef,
     data: ShareableRef
   ) => void;
-  const _scheduleOnJS: (fun: ShareableRef, args?: ShareableRef) => void;
-  const _updatePropsPaper: (
+  var _scheduleOnJS: (fun: ShareableRef, args?: ShareableRef) => void;
+  var _updatePropsPaper: (
     tag: number,
     name: string,
     updates: StyleProps | AnimatedStyle
   ) => void;
-  const _updatePropsFabric: (
+  var _updatePropsFabric: (
     shadowNodeWrapper: ShadowNodeWrapper,
     props: StyleProps | AnimatedStyle
   ) => void;
-  const _removeShadowNodeFromRegistry: (viewTag: number) => void;
-  const _measure: (viewTag: number) => MeasuredDimensions;
-  const _scrollTo: (
+  var _removeShadowNodeFromRegistry: (viewTag: number) => void;
+  var _measure: (viewTag: number) => MeasuredDimensions;
+  var _scrollTo: (
     viewTag: number,
     x: number,
     y: number,
     animated: boolean
   ) => void;
-  const _dispatchCommand: (
-    shadowNodeWrapper: ShadowNodeWrapper,
+  var _dispatchCommand: (
+    nodeRef: ShadowNodeWrapper | number,
```

Comment:
Please pay attention to this as I'm not 100% certain it's safe to change the name of the prop for a more logical one here.

Reply:
As discussed offline, `nodeRef` is a `number` on Paper and a `ShadowNodeWrapper` on Fabric, so I'm not sure if it's possible to type it properly since architecture detection happens in the runtime.

Reply:
I am not certain what the issue is tho. Here we declare that `nodeRef` is one of the types. I agree that if we are on Fabric passing a number as its value is illegal but I think that's still better than going for `unknown` or the like.

If it was desired we could differentiate for those nodes, for example:
```diff
-  shadowNodeWrapper: ShadowNodeWrapper,
+  paperRef: number,
+  fabricRef: ShadowNodeWrapper
```

and then check in runtime which prop should we use based on used architecture.

Still, we have to do something about it because this gives us type errors and I think it's the best approach for now.

Reply:
I like this idea but we'd still have to use `| undefined` in both places and somehow prevent passing undefined as both arguments.

Reply:
@tomekzaw you could've just asked

```TypeScript
  var _dispatchCommand:
    | ((
        paperRef: number,
        fabricRef: never,
        commandName: string,
        args: Array<unknown>
      ) => void)
    | ((
        paperRef: never,
        fabricRef: ShadowNodeWrapper,
        commandName: string,
        args: Array<unknown>
      ) => void);
      ```

Reply:
Let's go for it then